pkg/profiling/profiling.go

```diff
@@ -0,0 +1,149 @@
+package profiling
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"path"
+	"runtime"
+	"runtime/pprof"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+const (
+	fifoDefaultMaxFileCount = 10
+	heapdumpSubfolderName   = "heapdump"
+)
+
+type fifoDir struct {
+	maxFileCount int
+	dirPath      string
+}
+
+func (fd fifoDir) Create(fileName string) (*os.File, error) {
+	err := os.MkdirAll(fd.dirPath, os.ModePerm)
+	if err != nil {
+		if !os.IsExist(err) {
+			return nil, errors.Wrapf(err, "creating directory: %s", fd.dirPath)
+		}
+	}
+
+	entries, err := ioutil.ReadDir(fd.dirPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "reading directory: %s", fd.dirPath)
+	}
+
+	if len(entries) >= fd.maxFileCount {
+		var oldestEntryIndex int
+
+		for i, e := range entries {
+			oldestEntryInfo := entries[oldestEntryIndex]
+
+			if e.ModTime().Before(oldestEntryInfo.ModTime()) {
+				oldestEntryIndex = i
+			}
+		}
+
+		rmPath := path.Join(fd.dirPath, entries[oldestEntryIndex].Name())
+		os.Remove(rmPath)
```

Comment:
If I'm reading this right, this only removes a single file. I think we should keep deleting the oldest one until we get below `fd.maxFileCount`.

Reply:
You're right good catch. This only works correctly when starting with an empty directory. I changed it to sorting, which fortunately also made the code more simple :)